app/collins/controllers/actions/ipmi/GetPoolsAction.scala

```diff
@@ -0,0 +1,86 @@
+package collins.controllers.actions.ipmi
+
+import scala.concurrent.Future
+
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import play.api.libs.json.JsArray
+import play.api.libs.json.JsNumber
+import play.api.libs.json.JsObject
+import play.api.libs.json.JsString
+
+import collins.controllers.ResponseData
+import collins.controllers.SecureController
+import collins.controllers.actions.RequestDataHolder
+import collins.controllers.actions.SecureAction
+import collins.controllers.actions.ipaddress.AddressActionHelper
+import collins.models.IpmiInfo
+import collins.models.Truthy
+import collins.models.shared.AddressPool
+import collins.util.security.SecuritySpecification
+
+// Get pools, all or just those in use
+case class GetPoolsAction(
+  allPools: Truthy,
+  spec: SecuritySpecification,
+  handler: SecureController
+) extends SecureAction(spec, handler) with AddressActionHelper {
+
+  case class ActionDataHolder(all: Boolean) extends RequestDataHolder
+
+  override def validate(): Validation = Right(ActionDataHolder(allPools.toBoolean))
+
+  override def execute(rd: RequestDataHolder) = Future {
+    rd match {
+      case ActionDataHolder(all) =>
+        val pools: Set[String] =
+          if (all) {
+            IpmiInfo.AddressConfig.map(_.poolNames).getOrElse(Set())
```

Comment:
both arms of the conditional seem the same. Whats the purpose of the all param then?